protostar/commands/test/test_command.py

```diff
@@ -177,10 +177,10 @@ async def run(self, args) -> TestingSummary:
         for failed_test in summary.failed + summary.broken + summary.broken_suites:
             if hasattr(failed_test, "test_case_name"):
                 failed_test_cases.append(
-                    (str(failed_test.file_path), failed_test.test_case_name)
+                    (str(failed_test.file_path), failed_test.test_case_name)  # type: ignore
                 )
             elif hasattr(failed_test, "test_case_names"):
-                for test_name in failed_test.test_case_names:
+                for test_name in failed_test.test_case_names:  # type: ignore
```

Comment:
I suppressed the `pyright` errors here:
```
test_command.py:180:62 - error: Cannot access member "test_case_name" for type "BrokenTestSuiteResult"
test_command.py:183:46 - error: Cannot access member "test_case_names" for type "BrokenTestCaseResult"
test_command.py:183:46 - error: Cannot access member "test_case_names" for type "FailedTestCaseResult"
```
I think the solution with `hasattr` is less error-prone, clearer, and easier to maintain than handling every class separately.

Reply:
- "error-prone"? How so? I'd say this approach increases the feedback loop in case of errors, because we need to rely on tests more heavily (which this PR lacks btw)
- "clearer"? `# type: ignore` doesn't make it clearer. Type checker limits language functionalities, forcing developer to solve a problem using simpler tools or by rethinking the design, e.g. why do you append broken tests to `failed_test_cases`? Maybe a new `TestResult` type needs to be introduced?
- "easier to maintain than handling every class separately" Typed code is easier to maintain because it reduces the feedback loop, makes code spelunking and makes refactoring easier and makes jumping between files faster. The problem is with handling classes separately rather than with typing.

Reply:
Hmm, yes I first thought that it just would be simpler and safer to check the existence of the fields, but you've made some valid points, I also reconsidered this case and I think you're right. I'll go with handling the classes instead of reflection.

Reply:
Done in [use isinstance instead of hasattr](https://github.com/software-mansion/protostar/pull/895/commits/0e6d24325c9f01ee4de9296831c5fba501be56d8)